src/dft/backends/cufft/commit.cpp

```diff
@@ -138,27 +139,83 @@ class cufft_commit final : public dft::detail::commit_impl<prec, dom> {
         const int rank = static_cast<int>(config_values.dimensions.size());
-        const int istride = static_cast<int>(config_values.input_strides.back());
-        const int ostride = static_cast<int>(config_values.output_strides.back());
-        const int batch = static_cast<int>(config_values.number_of_transforms);
-        const int fwd_dist = static_cast<int>(config_values.fwd_dist);
-        const int bwd_dist = static_cast<int>(config_values.bwd_dist);
-        std::array<int, max_supported_dims> inembed;
-        if (rank == 2) {
-            inembed[1] = config_values.input_strides[1];
+        std::vector<int> inembed = { config_values.input_strides.begin(),
+                                     config_values.input_strides.end() };
+        std::vector<int> onembed = { config_values.output_strides.begin(),
+                                     config_values.output_strides.end() };
+        auto i_min = std::min_element(inembed.begin() + 1, inembed.end());
+        auto o_min = std::min_element(onembed.begin() + 1, onembed.end());
+        if constexpr (dom == dft::domain::REAL) {
+            if (i_min != inembed.begin() + rank) {
+                throw mkl::unimplemented(
+                    "dft/backends/cufft", __FUNCTION__,
+                    "cufft requires the last input stride to be the the smallest one for real transforms!");
+            }
+            if (o_min != onembed.begin() + rank) {
+                throw mkl::unimplemented(
+                    "dft/backends/cufft", __FUNCTION__,
+                    "cufft requires the last output stride to be the the smallest one for real transforms!");
+            }
         }
-        else if (rank == 3) {
-            inembed[2] = config_values.input_strides[2];
-            inembed[1] = config_values.input_strides[1] / inembed[2];
+        else {
+            if (o_min - onembed.begin() != i_min - inembed.begin()) {
+                throw mkl::unimplemented(
+                    "dft/backends/cufft", __FUNCTION__,
+                    "cufft requires that if  ordered by stride length, the order of strides is the same for input and output strides!");
+            }
         }
-        std::array<int, max_supported_dims> onembed;
-        if (rank == 2) {
-            onembed[1] = config_values.output_strides[1];
+        const int istride = static_cast<int>(*i_min);
+        const int ostride = static_cast<int>(*o_min);
+        inembed.erase(i_min);
+        onembed.erase(o_min);
+        if (o_min - onembed.begin() != rank) {
+            // swap dimensions to have the last one have the smallest stride
+            std::swap(n_copy[o_min - onembed.begin() - 1], n_copy[rank - 1]);
         }
-        else if (rank == 3) {
-            onembed[2] = config_values.output_strides[2];
-            onembed[1] = config_values.output_strides[1] / onembed[2];
+        for (int i = 0; i < rank - 1; i++) {
```

Comment:
```suggestion
        for (int i = 1; i < rank; i++) {
```
then remove the +1 from all the indexes in the loop.

Reply:
done